### DISCUSSION TITLE
Reduce code using UI files

### DISCUSSION THREAD

**Discussion by hsbasu** [amit9838/mousam#87] [Ideas]
Currently, the code has become massive as all the ui related codes are hard coded into the source. Use of .ui files built by [Glade](https://flathub.org/apps/org.gnome.Glade) or [Cambalache](https://flathub.org/apps/ar.xjuan.Cambalache) can reduce the code size. This would also reduce maintainability in the long terms.

**Comment by amit9838** [comment]
Actually, I haven't worked on the .ui thing. and I think its good for a python developer if the whole codebase is in python. 
And also migrating the whole ui, is not a good idea i guess🙂

**Reply by amit9838** [reply]
May be we can modularise the code and seperate it into backend and frontend, so that we can use the same backend with other ui framework for other platforms like windows

**Reply by hsbasu** [reply]
Have you checked [this](https://python-gtk-3-tutorial.readthedocs.io/en/latest/builder.html)? They explained how these ui files are independent of programming language and other advantages.

Actually, what I have observed is that it's pretty common in large app projects to use these ui files. Open any [gnome app source](https://gitlab.gnome.org/GNOME), there is a high probability that they are using ui files in `data/ui`.

**Reply by amit9838** [reply]
You are right. but here we are using various graphics built from scrach, and I think its not a good idea to rewrite and connect everything again. And also there is no issue with the performance. On the top of that i'm more comfortable with current codebase so lets keep it as it is.
